vendor/plist_to_html/plist_to_html/PlistToHtml.py

```diff
@@ -0,0 +1,208 @@
+            'BUG_VIEWER': 'bugviewer.js'
+        }
+
+        # Get the HTML layout file content.
+        self._layout = get_file_content(
+            os.path.join(self.layout_dir, 'layout.html'))
+
+        # Get the content of the HTML layout dependecies.
+        self._tag_contents = {}
+        for tag in self._layout_tag_files:
+            self._tag_contents[tag] = get_file_content(
+                os.path.join(self.layout_dir, self._layout_tag_files[tag]))
+
+            self._layout = self._layout.replace('<${0}$>'.format(tag),
+                                                self._tag_contents[tag])
+
+    def create(self, output_path, report_data):
+        """
+        Create html file with the given report data to the output path.
+        """
+        content = self._layout.replace('<$REPORT_DATA$>',
+                                       json.dumps(report_data))
+
+        with open(output_path, 'w+') as html_output:
+            html_output.write(content)
+
+
+def get_report_data_from_plist(plist):
+    files = plist['files']
+
+    reports = []
+    file_sources = {}
+    for diag in plist['diagnostics']:
+        bug_path_items = [item for item in diag['path']]
+        events = [i for i in bug_path_items if i.get('kind') == 'event']
+
+        report = []
+        for index, event in enumerate(events):
+            file_id = event['location']['file']
+            if file_id not in file_sources:
+                file_path = files[file_id]
+                source_data = open(file_path, 'r')
+                file_sources[file_id] = {'id': file_id,
+                                         'path': file_path,
+                                         'content': source_data.read()}
+
+            report.append({'line': event.location['line'],
+                           'col':  event.location['col'],
+                           'file': event.location['file'],
+                           'msg':  event.message,
+                           'step': index + 1})
+        reports.append(report)
+
+    return {'files': file_sources,
+            'reports': reports}
+
+
+def plist_to_html(file_path, output_path, html_builder):
+    """
+    Prints the results in the given file to HTML file.
+    """
+    if not file_path.endswith(".plist"):
+        print("Skipping input file '" + file_path + "' as it is not a plist.")
+        return
+
+    print("Parsing input file '" + file_path + "'")
+
+    try:
+        plist = plistlib.readPlist(file_path)
+
+        report_data = get_report_data_from_plist(plist)
+
+        if report_data is None or not len(report_data['reports']):
+            print('No report data in ' + file_path + ' file.')
+            return
+
+        html_filename = os.path.basename(file_path) + '.html'
+        html_output_path = os.path.join(output_path, html_filename)
+        html_builder.create(html_output_path, report_data)
+
+        print('Html file was generated: ' + html_output_path)
```

Comment:
It would be nice to print out here to view the results in the browser run `firefox html_output_path`

Reply:
Why do we want to print out this? It's obvious to view HTML files in a browser. Why do we want to prefer `firefox`?

Reply:
Otherwise, I change the code and the result looks like this:
```bash
Previous analysis results in '/home/username/.codechecker/html' have been removed, overwriting with current results.

Parsing input file '/home/username/.codechecker/tinyxml/xmltest.cpp_364275456d0df014e49eb680ef13ff72.plist'
Html file was generated: /home/username/.codechecker/html/xmltest.cpp_364275456d0df014e49eb680ef13ff72.plist.html
To view the results in a browser run:
> firefox /home/username/.codechecker/html/xmltest.cpp_364275456d0df014e49eb680ef13ff72.plist.html

Skipping input file /home/username/.codechecker/tinyxml/compile_cmd.json as it is not a plist.
```

Reply:
I've put the comment to the wrong place in the source, that is why it was misleading, sorry.
I thought to print out the directory with the html reports after all of the reports were processed at the end of the `PlistToHtml.py` script.